diffsims/utils/sim_utils.py

```diff
@@ -167,16 +169,20 @@ def get_vectorized_list_for_atomic_scattering_factors(
         Debye-Waller factors for each atom in the structure.
     """
 
-    scattering_params_dict = get_scattering_params_dict(scattering_params)
+    if scattering_params is not None:
+        scattering_params_dict = get_scattering_params_dict(scattering_params)
+    else:
+        scattering_params_dict = {}
 
     n_structures = len(structure)
     coeffs = np.empty((n_structures, 5, 2))
     fcoords = np.empty((n_structures, 3))
     occus = np.empty(n_structures)
     dwfactors = np.empty(n_structures)
+    default = np.zeros((5, 2))
 
     for i, site in enumerate(structure):
-        coeffs[i] = scattering_params_dict[site.element]
+        coeffs[i] = scattering_params_dict.get(site.element, default)
```

Comment:
What are these changes about?

Reply:
This is to deal with the case where `None` is passed as scattering params. To avoid having to write separate cases with lots of if-else statements, if the scattering params dict is empty or does not contain the element it just puts in 0's for the coefficients.  This is to avoid having to branch more than necessary in `get_kinematical_intensities`

Reply:
Yes, sorry this was me being slow, that's clever